checker/src/org/checkerframework/checker/nullness/qual/NonNull.java

```diff
@@ -48,16 +45,14 @@
  * @checker_framework.manual #nullness-checker Nullness Checker
  */
 @SubtypeOf(MonotonicNonNull.class)
-@ImplicitFor(types = { TypeKind.PACKAGE },
-    typeClasses = { AnnotatedPrimitiveType.class, AnnotatedNoType.class },
-    trees = {
-        Tree.Kind.NEW_CLASS,
-        Tree.Kind.NEW_ARRAY,
-        Tree.Kind.PLUS, // for String concatenation
-        // All literals except NULL_LITERAL:
-        Tree.Kind.BOOLEAN_LITERAL, Tree.Kind.CHAR_LITERAL,
-        Tree.Kind.DOUBLE_LITERAL, Tree.Kind.FLOAT_LITERAL,
-        Tree.Kind.INT_LITERAL, Tree.Kind.LONG_LITERAL, Tree.Kind.STRING_LITERAL })
+@ImplicitFor(types = { TypeKind.PACKAGE,
+                       TypeKind.INT, TypeKind.BOOLEAN, TypeKind.CHAR,
+                       TypeKind.DOUBLE, TypeKind.FLOAT, TypeKind.LONG,
+                       TypeKind.SHORT, TypeKind.BYTE },
+    // All literals except NULL_LITERAL:
+    literals = { LiteralKind.BOOLEAN, LiteralKind.CHAR, LiteralKind.DOUBLE,
```

Comment:
>Why is there duplication between the "types" and the "literals" now?

This was the previous behavior via `typeClasses = AnnotatedPrimitiveType.class`.  I think you are right that the primitive LiteralKinds are not necessary.  I'll make that change.

>The only item in literals that is not in types is LiteralKind.STRING. Can that be replaced with a TypeKind.STRING instead?

No, because local variables that are Strings should be defaulted to `@Nullable`